lib/ansible/modules/cloud/amazon/iam_group.py

```diff
@@ -0,0 +1,376 @@
+    if group is None:
+        try:
+            group = connection.create_group(**params)
+            changed = True
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    # Manage managed policies
+    current_attached_policies = get_attached_policy_list(connection, params['GroupName'])
+    if not compare_attached_group_policies(current_attached_policies, managed_policies):
+        # If managed_policies has a single empty element we want to remove all attached policies
+        if len(managed_policies) == 1 and managed_policies[0] == "":
+            for policy in current_attached_policies:
+                try:
+                    connection.detach_group_policy(GroupName=params['GroupName'], PolicyArn=policy['PolicyArn'])
+                except (ClientError, ParamValidationError) as e:
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        # Detach policies not present
+        current_attached_policies_arn_list = []
+        for policy in current_attached_policies:
+            current_attached_policies_arn_list.append(policy['PolicyArn'])
+
+        for policy_arn in list(set(current_attached_policies_arn_list) - set(managed_policies)):
+            try:
+                connection.detach_group_policy(GroupName=params['GroupName'], PolicyArn=policy_arn)
+            except (ClientError, ParamValidationError) as e:
+                module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+
+        # If there are policies in managed_policies attach each policy
+        if managed_policies != [None]:
+            for policy_arn in managed_policies:
+                try:
+                    connection.attach_group_policy(GroupName=params['GroupName'], PolicyArn=policy_arn)
+                except (ClientError, ParamValidationError) as e:
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        changed = True
+
+    # Manage group memberships
+    current_group_members = get_group(connection, params['GroupName'])['Users']
+    current_group_members_list = []
+    for member in current_group_members:
+        current_group_members_list.append(member['UserName'])
+
+    if not compare_group_members(current_group_members_list, users):
+
+        # If users has a single empty element we want to remove all users that are members of the group
+        if len(users) == 1 and users[0] is None:
+            for user in current_group_members_list:
+                try:
+                    connection.remove_user_from_group(GroupName=params['GroupName'], UserName=user)
+                except (ClientError, ParamValidationError) as e:
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        for user in list(set(current_group_members_list) - set(users)):
+            try:
```

Comment:
But this step is not skipped. And all users from the group is removed each time when we run playbook. Because `current_group_members_list` has a list of users from aws. And `users` is an empty list.  And `current_group_members_list` - empty list = `current_group_members_list`.